query-engine/query-engine-node-api/src/node_drivers/engine.rs

```diff
@@ -0,0 +1,140 @@
+use crate::engine::QueryEngine;
+
+#[allow(unused_imports)]
+use napi::bindgen_prelude::Promise as JsPromise;
+
+use napi::{Env, JsFunction, JsObject, JsString, JsUnknown};
+use napi_derive::napi;
+
+pub struct NodejsFunctionContext {
+    env: Env,
+    pub query_raw: JsFunction,
+    pub query_raw_typed: JsFunction,
+    pub execute_raw: JsFunction,
+    pub execute_raw_typed: JsFunction,
+    pub version: JsFunction,
+}
+
+impl NodejsFunctionContext {
+    pub fn query_raw(&self) -> napi::Result<()> {
+        let arg = self.env.create_string("SELECT * FROM users").unwrap();
+        let _promise: JsUnknown = self.query_raw.call(None, &[arg])?;
+        // TODO: cast JsUnknown to JsPromise
+
+        Ok(())
+    }
+
+    pub fn query_raw_typed(&self) -> napi::Result<()> {
+        let arg = self.env.create_string("SELECT * FROM users").unwrap();
+        let _promise: JsUnknown = self.query_raw_typed.call(None, &[arg])?;
+        // TODO: cast JsUnknown to JsPromise, idk how yet :/
+
+        Ok(())
```

Comment:
*Be careful, goodly wanderer who shall traverse this path*: the `JsValue` <-> `Rust type` conversion APIs seem much more painful in napi.rs than they are in `wasm_bindgen`. Good luck